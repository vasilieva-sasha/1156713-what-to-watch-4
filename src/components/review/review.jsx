import React from "react";
import PropTypes from "prop-types";

const Review = (props) => {
  const {review} = props;
  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{review.text}</p>

        <footer className="review__details">
          <cite className="review__author">{review.author}</cite>
          <time className="review__date" dateTime={review.date}>{review.date}</time>
        </footer>
      </blockquote>

      <div className="review__rating">{review.rating}</div>
    </div>
  );
};

Review.propTypes = {
  review: PropTypes.shape({
    id: PropTypes.number.isRequired,
    author: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    rating: PropTypes.string.isRequired,
  })
};

export {Review};
