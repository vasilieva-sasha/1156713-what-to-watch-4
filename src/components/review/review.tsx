import * as React from "react";
import * as moment from "moment";
import {ReviewInterface} from "../../types";

interface Props {
  review: ReviewInterface;
}

const Review: React.FunctionComponent<Props> = (props: Props) => {
  const {review} = props;
  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{review.comment}</p>

        <footer className="review__details">
          <cite className="review__author">{review.user.name}</cite>
          <time className="review__date" dateTime={review.date}>{moment(review.date).format(`MMMM DD, YYYY`)}</time>
        </footer>
      </blockquote>

      <div className="review__rating">{review.rating}</div>
    </div>
  );
};

export default Review;
