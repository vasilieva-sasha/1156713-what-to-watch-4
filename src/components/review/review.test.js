import React from "react";
import renderer from "react-test-renderer";
import {Review} from "./review.jsx";

const mock = {
  id: 0,
  author: ``,
  text: ``,
  date: ``,
  rating: ``,
};

describe(`ReviewComponent`, () => {
  it(`review component renders correctly`, () => {
    const tree = renderer.create(
        <Review review={mock} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
