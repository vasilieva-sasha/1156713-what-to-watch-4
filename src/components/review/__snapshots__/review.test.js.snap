// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewComponent review component renders correctly 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        
      </cite>
      <time
        className="review__date"
        dateTime=""
      >
        
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    
  </div>
</div>
`;
